### ISSUE TITLE
Guest-specific operations were attempted on a machine that is not ready for guest communication: Error Observed

### ISSUE THREAD

**Issue by skorada-rvbd** [josenk/vagrant-vmware-esxi#23]
Hi Jonathan,

I observed the issue when bringing up VMs using the plugin by triggering a script.

```
RUBY_PLATFORM: x86_64-linux
==> test-esx-vm: --- ESXi version    : VMware ESXi 6.5.0 build-4887370
Bringing machine 'test-esx-vm' up with 'vmware_esxi' provider...
==> test-esx-vm: Virtual Machine will be built.
==> test-esx-vm: --- Avail DS vols   : ["datastore1"]
==> test-esx-vm: --- WARNING         : esxi_disk_store not set, using "--- Least Used ---"
==> test-esx-vm: --- Avail Networks  : ["VM Network", "Management Network", "", "internal"]
==> test-esx-vm: --- Network[0]      : VM Network
==> test-esx-vm: ---   --- ESXi Summary ---
==> test-esx-vm: --- ESXi host       : 10.xx.xx.xx
==> test-esx-vm: --- Virtual Network : ["VM Network"]
==> test-esx-vm: --- Disk Store      : datastore1
==> test-esx-vm: --- Resource Pool   : /
==> test-esx-vm: ---  --- Guest Summary ---
==> test-esx-vm: --- VM Name         : test-esx-vm
==> test-esx-vm: --- Box             : bento/ubuntu-16.04
==> test-esx-vm: --- Box Ver         : 201801.02.0
==> test-esx-vm: --- Memsize (MB)    : 6000
==> test-esx-vm: --- CPUS            : 2
==> test-esx-vm: --- Guest OS type   : ubuntu-64
==> test-esx-vm: ---   --- Guest Build ---
VMware ovftool 4.2.0 (build-5965791)
ovftool command: ovftool --noSSLVerify   --net:"net0=VM Network" -dm=thin  -ds="datastore1" --name="test-esx-vm" "/home/hostname/.vagrant.d/boxes/bento-VAGRANTSLASH-ubuntu-16.04/201801.02.0/vmware_desktop/ZZZZ_test-esx-vm.vmx" vi://root:******@10.xx.xx.xx/
Opening VMX source: /home/sdptest/.vagrant.d/boxes/bento-VAGRANTSLASH-ubuntu-16.04/201801.02.0/vmware_desktop/ZZZZ_test-esx-vm.vmx
Opening VI target: vi://root@10.xx.xx.xx:443/
Deploying to VI: vi://root@10.xx.xx.xx:443/
Transfer Completed
Completed successfully
==> test-esx-vm: --- Keeping file    : /home/hostname/.vagrant.d/boxes/bento-VAGRANTSLASH-ubuntu-16.04/201801.02.0/vmware_desktop/ZZZZ_test-esx-vm.vmx
==> test-esx-vm: --- VMID            : 327
==> test-esx-vm: --- Custom vmx      : Append vhv.enable = "TRUE"
==> test-esx-vm: --- VM has been Powered On...
==> test-esx-vm: --- Waiting for state "running"
==> test-esx-vm: --- Success, state is now "running"
num adapters: 1,  vm.network.count: 0
    test-esx-vm:
    test-esx-vm: Vagrant insecure key detected. Vagrant will automatically replace
    test-esx-vm: this with a newly generated keypair for better security.
    test-esx-vm:
    test-esx-vm: Inserting generated public key within guest...
    test-esx-vm: Removing insecure key from the guest if it's present...
    test-esx-vm: Key inserted! Disconnecting and reconnecting using new SSH key...
==> test-esx-vm: Rsyncing folder: /home/hostname => /vagrant
==> test-esx-vm: Machine not provisioned because `--no-provision` is specified.

RUBY_PLATFORM: x86_64-linux
==> test-esx-vm: --- ESXi version    : VMware ESXi 6.5.0 build-4887370
==> test-esx-vm: powered off
==> test-esx-vm: Virtual Machine is already built.
==> test-esx-vm: --- VM has been Powered On...
num adapters: 1,  vm.network.count: 0
**Guest-specific operations were attempted on a machine that is not
ready for guest communication. This should not happen and a bug
should be reported.**
```

Script does the following:
```
vagrant up --provider=vmware_esxi --no-provision
#some operation to get the IP of the VM (vagrant ssh-config and obtain IP)
sleep 10
vagrant reload --provision
```

Vagrantfile:

```
#TODO: Configure private networks and assign static ip
VAGRANTFILE_API_VERSION = "2"
Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.ssh.forward_agent = true
  config.vm.define test-esx-vm do |x|
    x.vm.box = "bento/ubuntu-16.04"
    x.vm.provider :vmware_esxi do |v|
      v.esxi_hostname = 10.xx.xx.xx
      v.esxi_username = "root"
      v.esxi_password = "password"
      v.guest_custom_vmx_settings = [['vhv.enable', 'TRUE']]
      v.guest_name = test-esx-vm
      v.guest_memsize = 6000
      v.guest_numvcpus = 2
      v.esxi_virtual_network = ["VM Network"]
      v.debug = "true"
    end
    x.vm.provision :shell do |s|
       s.path = /home/script.sh
    end
  end
end
```

Version
```

vagrant-vmware-esxi (2.0.6)
```

**Comment by skorada-rvbd** [comment]
I observed that on increasing the sleep to 30s I dont face the issue anymore.
Not really sure what caused the issue in the first place

vagrant up --provider=vmware_esxi --no-provision
#some operation to get the IP of the VM (vagrant ssh-config and obtain IP)
sleep 30
vagrant reload --provision

**Comment by josenk** [comment]
I found a problem in action.rb code.   The code to "vagrant reload" will halt the system then up.   But it doesn't wait for any state...  I took another look at the lib-virt and virtualbox plugin.   They both wait for running state and then does provision.   I'll duplicate that behaviour and your problem should be fixed.

https://github.com/josenk/vagrant-vmware-esxi/blob/v2.0.6/lib/vagrant-vmware-esxi/action.rb
